### PR TITLE
[VTA][Relay] Extending Vision model coverage compilation for VTA

### PR DESCRIPTION
**Pull request by tmoreau89**
This fixes Relay compilation for ResNet18_v2, ResNet34_v2,  ResNet50_v2, ResNet101_v2, ResNet152_v2, Alexnet, VGG11, VGG13, VGG16, VGG19.

Currently the following models will work in HW/SIM (white-listed in tutorial until more models are supported): ResNet18_v2, ResNet34_v2,  ResNet50_v2, ResNet101_v2

Performance on `pynq` IoT platform: 371.68ms, 470.86ms, 662.13ms and 916.26ms for ResNet-18, 34, 50, and 101 respectively.

Bug fixes:
- Elementwise multiplication support in graph pack (layout modification)
- Maxpooling on NCHWnc format
- Reducing the micro-op/insn buffer size to `1<<24` to avoid out of memory errors on the pynq

[resolved]: getting quantization to pass on these models is pending on #3543 being merged. I've addressed this dependence with a TODO.

### CODE REVIEW

File: vta/tutorials/frontend/deploy_resnet_on_vta.py [hunk 1]
```diff
@@ -61,6 +62,8 @@
 # Make sure that TVM was compiled with RPC=1
 assert tvm.module.enabled("rpc")
 
+# Increase python recursion limit to traverse Relay program
+sys.setrecursionlimit(10000)
```
Comment (tmoreau89) [inline vta/tutorials/frontend/deploy_resnet_on_vta.py]: @jroesch is there a better place to put this line?

**Comment by MarisaKirisame** [comment]
@tmoreau89 maybe we should set limit in tvm __init__.py

**Comment by tmoreau89** [comment]
@MarisaKirisame If we set it in init.py, then do we not revert it back to its original value??

**Comment by MarisaKirisame** [comment]
@tmoreau89 why revert at all? tvm is supposed to be used with a high recursion limit. if user wish otherwise she should pick a better constant herself.

**Comment by tmoreau89** [comment]
Ok, which init.py should we include the lines in? the top level?

**Comment by MarisaKirisame** [comment]
@tmoreau89 I think relay is fine for now. If tvm also need it, we can move it up lateron.

File: vta/tutorials/frontend/deploy_vision_on_vta.py [hunk 2]
```diff
@@ -100,7 +115,7 @@
     # the host, make sure you've set the variables below to the IP of
     # your board.
     device_host = os.environ.get("VTA_PYNQ_RPC_HOST", "192.168.2.99")
-    device_port = os.environ.get("VTA_PYNQ_RPC_PORT", "9091")
+    device_port = int(os.environ.get("VTA_PYNQ_RPC_PORT", "9091"))
     if not tracker_host or not tracker_port:
         remote = rpc.connect(device_host, int(device_port))
```
Comment (liangfu) [inline vta/tutorials/frontend/deploy_vision_on_vta.py]: why convert `device_port` to int again?

Comment (tmoreau89) [reply vta/tutorials/frontend/deploy_vision_on_vta.py]: that's a carry over from old code and should be cleaned up, thanks for the catch!

**Pull request merged**